### PR TITLE
feat: allow to pass options on per-file basis

### PR DESCRIPTION
**Pull request by jvolker**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -52,14 +52,26 @@ export default function (userOptions) {
...
         // Iterate over all option sets.
-        return optionsList.reduce((stepSequence, options) => {
+        return combinedOptionsList.reduce((stepSequence, options) => {
           const stepOptions = {
             ...defaultOptions,
             ...options
           }
 
           if (!minimatch(filename, stepOptions.src)) {
+          // if (((!minimatch(filename, stepOptions.src) && !file.sharp) || file.sharp)) {
```
Comment (axe312ger) [inline src/index.js]: you can remove this comment :)

File: test/index.test.js [hunk 2]
```diff
@@ -121,6 +121,43 @@ test.cb('test method without arguments', (t) => {
     })
 })
 
+test.cb('test method without arguments - options attached to file', (t) => {
+  const { metalsmith } = t.context
+
+  metalsmith
+    .use((files) => {
+      files['example.jpg'].sharp = {
+        namingPattern: '{dir}negated{ext}',
```
Comment (axe312ger) [inline test/index.test.js]: Lets use a different naming scheme here, you are actually overwriting the file negated file from the other test. You might even later on test the file from the other test and your's didn't work at all 😱

**Review by axe312ger — CHANGES_REQUESTED** [review]
Well done, a few small comments I made. Afterwards we can merge this and push a new version :)

**Comment by axe312ger** [comment]
@jvolker two small comments and you need to resolve the conflicts. Make sure to rebase on the next branch which contains the new version including the change from #71 

Will then do a few changes to the used dependencies (update babel to use the env preset etc)  and push out a new version including the multimatch and your config per file basis :)

File: src/index.js (hunk 1 above)
Comment (jvolker) [reply src/index.js]: 😬

File: test/index.test.js (hunk 2 above)
Comment (jvolker) [reply test/index.test.js]: Good point. 👍

**Comment by jvolker** [comment]
Thanks for the feedback. I'm working on it.

**Pull request merged**